### PR TITLE
chore: update workflow

### PR DESCRIPTION
**Pull request by lattejed**
## Motivation

The `cargo test` command was relaxed when being ported to the workflow from the original monorepo, causing tests behind feature flags to be skipped.

Ref https://github.com/nomad-xyz/rust/issues/118

NOTE:

**Requires https://github.com/nomad-xyz/rust/pull/165 to be pushed first or all tests will start failing**

## Solution

Make the `cargo test` and `fmt`, `clippy` and `build` commands more explicit

### CODE REVIEW

**Comment by lattejed** [comment]
FYI, tests should be failing because https://github.com/nomad-xyz/rust/pull/165 hasn't been merged yet

**Comment by lattejed** [comment]
Still getting failures here after the merge, I'm looking into it

**Comment by lattejed** [comment]
This should be good to go now

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -49,10 +49,10 @@ jobs:
       - uses: Swatinem/rust-cache@v1
 
       - name: Rustfmt
-        run: cargo fmt -- --check
+        run: cargo fmt --all -- --check
```
Comment (ltchang2019) [inline .github/workflows/rust.yml:52]: Odd that --workspace doesn't work for fmt huh

**Review by ltchang2019 — APPROVED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (lattejed) [reply .github/workflows/rust.yml:52]: Yeah, strange they didn't deprecated it as well

**Pull request merged**